source/dofs/dof_handler_policy.cc

```diff
@@ -872,6 +872,15 @@ namespace internal
                                            fe_index_2 =
                                              line->nth_active_fe_index(g);
 
+                        // as described in the hp paper, we only unify on lines
+                        // when there are at most two different FE objects
+                        // assigned on it.
+                        // however, more than two 'active_fe_indices' can be
+                        // attached that still fulfill the above criterion,
+                        // e.g. when two different FE_Q(2) objects are assigned
+                        // to neighboring cells.
+                        // we check for this particular case by comparing the
+                        // number of 'dofs_per_line' of all active FE objects.
```

Comment:
This is probably a dumb question, but is this enough to establish that unification is possible? I.e., if I came up with another finite element that was not the same as FE_Q(2) but had the same dofs per object would this still combine their DoFs?

Reply:
@bangerth and I discussed about that and we both have similar doubts about the current implementation in the code.

I would like to suggest to try a `dynamic_cast` from one `FiniteElement` object to the other, or something similar. What do you both think about this alternative approach?

Reply:
No, dynamic_cast is not good enough. Consider this for the case of a vertex: there, a Q1 and a Q2 element will conceptually have separate separate (sets of) degrees of freedom, but in reality they map one-to-one, and we can give them identical numbers and don't need to worry about there being two sets of indices. You can probably construct similar cases if you have two elements on both sides of a face where one element is, say, a Q1 and the other is a Q1 enriched by a bubble function that is zero on the boundary.